### PR TITLE
 🔨 New settings-library for settings

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

NEW ``packages/settings-library`` package for settings : implements new settings designed drafted in [sample-app-settings](https://github.com/pcrespov/sample-app-settings) repo.

#### Highlights
- ``settings_library.base`` defines  new base class ``BaseCustomSettings``:
   - implements ``create_from_envs`` constructor
   - sets ``BaseCustomSettings.Config``
- ``settings_library.cli_utils`` defines a factory to create  settings commands  for typer CLI
- ``settings_library.celery, postgres, rabbit, redis, s3`` reimplements ``models_library.settings.*`` using new settings design
    - all fields are UPPER_CASE
    - Uses @cached_property for read-only composed fields (e.g. urls). See tests for more details 

#### In addition
- ADDS new  fixtures ``pytest_simcore.cli_runner`` 
- UPDATED ``storage`` service:  replaces ``model_library.settings`` by ``settings_library``
    - Adapts ``Settings``
    - Simplifies CLI  and improves tests 

## Related issue/s

This is **one of the four PR** issued to refactor ``settings``. The other are:
1. ~~#2369: refactors storage *(merged)*~~
2. #2376: removes trafaret and refactors webserver 
3.  Replaces  ``model_library.settings`` by ``settings_library`` in the entire repo

## How to test

```command line
make devenv
cd packages/settings-library
make install-dev
make tests

cd ...
cd services/storage
make install-dev
make tests
```

### CODE REVIEW

File: ci/github/unit-testing/settings-library.bash [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+install() {
+    bash ci/helpers/ensure_python_pip.bash
+    pushd packages/settings-library; pip3 install -r requirements/ci.txt; popd;
+    pip list -v
+}
+
+test() {
+    pytest --cov=settings_library --durations=10 --cov-append \
+          --color=yes --cov-report=term-missing --cov-report=xml --cov-config=.coveragerc \
+          -v packages/settings-library/tests
```
Comment (sanderegg) [inline ci/github/unit-testing/settings-library.bash:17]: are these tests real unit tests? if yes you can add the the pytest-xdist to use multiple CPUs

File: packages/settings-library/src/settings_library/postgres.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    # auth
+    POSTGRES_USER: str
+    POSTGRES_PASSWORD: SecretStr
+
+    # database
+    POSTGRES_DB: str = Field(..., description="Database name")
+
+    # pool connection limits
+    POSTGRES_MINSIZE: IntGe1 = Field(
+        1, description="Maximum number of connections in the pool"
```
Comment (sanderegg) [inline packages/settings-library/src/settings_library/postgres.py:27]: inverse the descriptions with max size

File: packages/settings-library/src/settings_library/postgres.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    )
+    POSTGRES_MAXSIZE: IntGe1 = Field(
+        50, description="Minimum number of connections in the pool"
+    )
+
+    POSTGRES_APPNAME: Optional[str] = Field(
```
Comment (sanderegg) [inline packages/settings-library/src/settings_library/postgres.py:33]: POSTGRES_CLIENT_NAME ?

**Review by sanderegg — APPROVED** [review]

File: ci/github/unit-testing/settings-library.bash (hunk 1 above)
Comment (pcrespov) [reply ci/github/unit-testing/settings-library.bash:17]: yes. pytest-xdist looks definitively useful but here tests take 6 secs. I rather do that in other larger test suite

File: packages/settings-library/src/settings_library/postgres.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import urllib.parse
+from functools import cached_property
+from typing import Optional
+
+from pydantic import Field, PostgresDsn, SecretStr, conint, validator
+
+from .base import BaseCustomSettings
+from .basic_types import PortInt
+
+IntGe1 = conint(ge=1)
```
Comment (GitHK) [inline packages/settings-library/src/settings_library/postgres.py:10]:
```suggestion
IntGE1 = conint(ge=1)
```

**Review by GitHK — APPROVED** [review]
Just a few minor comments.

File: packages/settings-library/tests/test__code_syntax.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# pylint:disable=unused-variable
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply packages/settings-library/tests/test__code_syntax.py:1]: The convention is: 
- every module has a test file associated. E.g. module "foo.bar.woo" has ``test_foo_bar_woo``
- for tests that are not associated to the modules, we use double ``_`` after test:, i.e. ``test__somename``

**Pull request merged**